frontend/test/src/container/SearchResult.spec.js

```diff
@@ -0,0 +1,130 @@
+  const updateSearchResultListCallBack = sinon.spy()
+  const setBreadcrumbsCallBack = sinon.spy()
+
+  const dispatchMock = (params) => {
+    if (isFunction(params)) return params(dispatchMock)
+    const { type } = params
+    switch (type) {
+      case `${UPDATE}/${SEARCH_RESULTS_LIST}`: updateSearchResultListCallBack(); break
+      case `${SET}/${BREADCRUMBS}`: setBreadcrumbsCallBack(); break
+    }
+    return params
+  }
+
+  const props = {
+    breadcrumbs: [],
+    searchResult: searchResult,
+    contentType: contentType,
+    user: user,
+    system: {
+      config: {
+        instance_name: 'instanceName'
+      }
+    },
+    t: key => key,
+    dispatch: dispatchMock,
+    registerCustomEventHandlerList: () => { },
+    registerLiveMessageHandlerList: () => { }
+  }
+
+  const SearchResultWithHOC = withRouterMock(SearchResultWithoutHOC)
+  const wrapper = shallow(<SearchResultWithHOC {...props} />).dive()
+  const searchResultInstance = wrapper.instance()
+
+  describe('TLM handlers', () => {
+    describe('eventType content', () => {
+      const tlmData = {
+        content: contentFromApi
+      }
+      describe('handleContentModifiedOrDeleted', () => {
+        searchResultInstance.handleContentModifiedOrDeleted(tlmData.content)
+        it('should call this.props.dispatch(updateUser())', () => {
+          expect(updateSearchResultListCallBack.called).to.equal(true)
+        })
+      })
+    })
+  })
+
+  describe('its internal function', () => {
+    describe('hasMoreResults', () => {
+      it("should return false if it doesn't have more results", () => {
+        wrapper.setState({ totalHits: 1 })
+        expect(searchResultInstance.hasMoreResults()).to.equal(false)
+      })
+
+      it("should return false if it doesn't have more results", () => {
+        wrapper.setState({ totalHits: 30 })
+        expect(searchResultInstance.hasMoreResults()).to.equal(true)
+      })
+    })
+
+    describe('getContentName', () => {
+      it('should return the label of the non-file type contents', () => {
+        const content = {
+          contentType: 'thread',
+          filename: 'filename.thread.html',
+          label: 'label'
+        }
+        expect(searchResultInstance.getContentName(content)).to.equal(content.label)
+      })
+
+      it('should return the filename of the content for a file type content', () => {
+        const content = {
+          contentType: 'file',
+          filename: 'filename.jpg',
+          label: 'label'
+        }
+        expect(searchResultInstance.getContentName(content)).to.equal(content.filename)
+      })
+    })
+
+    describe('getPath', () => {
+      it('should return the content path', () => {
+        const parentList = [
+          {
+            label: 'first'
+          },{
+            label: 'second'
+          }
+        ]
+        expect(searchResultInstance.getPath(parentList)).to.equal(
+          `${parentList[1].label} / ${parentList[0].label} / `
```

Comment:
if this test pass, I think there might be something wrong since an url cannot have spaces in it.
This indicates a potential bug in getPath or `.equal()` isn't the proper function to use

Reply:
This path is not the url, is the path inside Tracim (folder, subfolder,...) which is shown in the search

Reply:
oh I see